src/index.ts

```diff
@@ -423,7 +459,8 @@ export default class JSONFormatter {
       let index = 0;
       const addAChild = ()=> {
         const key = this.keys[index];
-        const formatter = new JSONFormatter(this.json[key], this.open - 1, this.config, key);
+        const displayKey = (this.isArrayRange ? (Number(this.key.split(RANGE_DELIMITER)[0]) + Number(key)).toString() : key);
+        const formatter = new JSONFormatter(this.getJSONForKey(key), this.open - 1, this.config, key, displayKey);
```

Comment:
Here we know that we're constructing an array range, maybe here pass an argument to constructor to explicitly specify that ti should be rendered as an array range instead of it depending on existence of a string in the key